src/cmap/auth/mongodb_aws.ts

```diff
@@ -2,44 +2,44 @@ import http = require('http');
 import crypto = require('crypto');
 import url = require('url');
 import * as BSON from '../../bson';
-import { AuthProvider } from './auth_provider';
+import { AuthProvider, AuthContext } from './auth_provider';
 import { MongoCredentials } from './mongo_credentials';
 import { MongoError } from '../../error';
 import { maxWireVersion } from '../../utils';
+import type { Callback, BSONSerializeOptions } from '../../types';
 
-let aws4: any;
-try {
-  aws4 = require('aws4');
-} catch (e) {
-  // don't do anything;
-}
+import { optionalRequire } from '../../deps';
+const aws4 = optionalRequire<typeof import('aws4')>('aws4');
```

Comment:
if `aws4` is not installed, what does `typeof import('aws4')` do?

Reply:
Nothing, its only a type annotation so that import is actually just referencing `@types/aws4` regardless of the actual aws4 being installed.